CAT/model/IRT.py

```diff
@@ -277,6 +284,86 @@ def get_fisher(self, student_id, question_id, pred_all):
         q = 1 - pred
         fisher_info = (q*pred*(alpha * alpha.T)).numpy()
         return fisher_info
+    def IRT_derivate(self,pred_all):
+
+        new_predictions = {}
+        for sid, qid_dict in pred_all.items():
+            new_predictions[sid] = {}
+            for qid, pred in qid_dict.items():
+                new_pred = pred * (1 - pred)
+                new_predictions[sid][qid] = new_pred
+
+    def bce_loss_derivative(self,pred, target):
+        derivative = (pred - target) / (pred * (1 - pred))
+        return derivative
+    def get_BE_weights(self, pred_all):
+        """
+        Returns:
+            predictions, dict[sid][qid]
```

Comment:
please write comments regarding your new functions, including description, args and returns